6.bulk_Morphology_Elastic_Network/3.model_coefficients/notebooks/2.model_coefficients_visualization.ipynb

```diff
@@ -0,0 +1,2439 @@
+                "dir.create(output_path, recursive = TRUE, showWarnings = FALSE)"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": 4,
+            "metadata": {
+                "vscode": {
+                    "languageId": "r"
+                }
+            },
+            "outputs": [],
+            "source": [
+                "## function to process the data for visualization\n",
+                "process_subset_data <- function(data_path){\n",
+                "    # read in the data\n",
+                "    data <- read.csv(data_path, header = TRUE, sep = \",\", stringsAsFactors = FALSE)\n",
+                "    # get the basename of the files\n",
+                "\n",
+                "    data <- data %>%\n",
+                "        dplyr::arrange(desc(abs(coefficients))) %>%\n",
+                "        tidyr::separate(\n",
+                "            feature_names,\n",
+                "            into = c(\n",
+                "                \"compartment\",\n",
+                "                \"feature_group\",\n",
+                "                \"measurement\",\n",
+                "                \"channel\", \n",
+                "                \"parameter1\", \n",
+                "                \"parameter2\"\n",
+                "            ),\n",
+                "            sep = \"_\",\n",
+                "            remove = FALSE\n",
+                "        ) %>%\n",
+                "        dplyr::mutate(channel_cleaned = channel) %>%\n",
+                "        dplyr::arrange(desc(abs(coefficients)))\n",
+                "\n",
+                "    # Clean channel for visualization\n",
+                "    data$channel_learned <- dplyr::recode(data$channel,\n",
+                "            \"CorrDNA\" = \"nuclei\",\n",
+                "            \"CorrMito\" = \"Mito\",\n",
+                "            \"CorrER\" = \"ER\",\n",
+                "            \"CorrGasdermin\" = \"gasdermin\",\n",
+                "            \"CorrPM\" = \"PM\",\n",
+                "            .default = \"other\",\n",
+                "            .missing=\"other\"\n",
+                "    )\n",
+                "    data <- data %>%\n",
+                "        dplyr::group_by(feature_group, channel_learned, compartment) %>%\n",
+                "        dplyr::slice_max(order_by = coefficients, n = 1)\n",
+                "    return(data)\n",
+                "}"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": 6,
+            "metadata": {
+                "vscode": {
+                    "languageId": "r"
+                }
+            },
+            "outputs": [
+                {
+                    "name": "stderr",
+                    "output_type": "stream",
+                    "text": [
+                        "Warning message:\n",
```

Comment:
Are these warnings important to take heed of and/or document in the notebook?

Reply:
In this case, I would say no. They are annoying and I need to remove them eventually...